src/helpers.js

```diff
@@ -57,10 +60,177 @@ export function idsToIgnoredRules(ruleIds = []) {
+    `Rule: \`${ruleId}\``,
+    rangeText,
+    '', '',
+    '<!-- If at all possible, please include code to reproduce this issue! -->',
+    '', '',
+    'Debug information:',
+    '```json',
+    JSON.stringify(await getDebugInfo(worker), null, 2),
+    '```'
+  ].join('\n'))
+  const newIssueURL = `${issueURL}?title=${title}&body=${body}`
+  return {
+    type: 'Error',
+    severity: 'error',
+    html: `${escapeHTML(titleText)}. See the trace for details. ` +
+      `<a href="${newIssueURL}">Report this!</a>`,
+    filePath,
+    range: rangeFromLineNumber(textEditor, 0),
+    trace: [
+      {
+        type: 'Trace',
+        text: `Original message: ${ruleId} - ${message}`,
+        filePath,
+        severity: 'info',
+      },
+      {
+        type: 'Trace',
+        text: rangeText,
+        filePath,
+        severity: 'info',
+      },
+    ]
+  }
+}
+
+export async function processMessages(response, textEditor, showRule, worker) {
+  return Promise.all(response.map(async ({
+    message, line, severity, ruleId, column, fix, endLine, endColumn
+  }) => {
+    const filePath = textEditor.getPath()
+    const textBuffer = textEditor.getBuffer()
+    let linterFix = null
+    if (fix) {
+      const fixRange = new Range(
+        textBuffer.positionForCharacterIndex(fix.range[0]),
+        textBuffer.positionForCharacterIndex(fix.range[1])
+      )
+      linterFix = {
+        range: fixRange,
+        newText: fix.text
+      }
+    }
+    const msgLine = line - 1
+    let msgCol
+    let msgEndLine
+    let msgEndCol
+    let eslintFullRange = false
+    if (typeof endColumn !== 'undefined' && typeof endLine !== 'undefined') {
+      eslintFullRange = true
+      // Here we always want the column to be a number
+      msgCol = Math.max(0, column - 1)
+      msgEndLine = endLine - 1
+      msgEndCol = endColumn - 1
+    } else {
+      // We want msgCol to remain undefined if it was initially so
+      // `rangeFromLineNumber` will give us a range over the entire line
+      msgCol = typeof column !== 'undefined' ? column - 1 : column
+    }
+
+    let ret
+    let range
+    try {
+      if (eslintFullRange) {
+        validatePoint(textEditor, msgLine, msgCol)
+        validatePoint(textEditor, msgEndLine, msgEndCol)
+        range = [[msgLine, msgCol], [msgEndLine, msgEndCol]]
+      } else {
+        range = rangeFromLineNumber(textEditor, msgLine, msgCol)
+      }
+      ret = {
+        filePath,
+        type: severity === 1 ? 'Warning' : 'Error',
+        range
+      }
+
+      if (showRule) {
+        const elName = ruleId ? 'a' : 'span'
+        const href = ruleId ? ` href=${ruleURI(ruleId).url}` : ''
+        ret.html = `<${elName}${href} class="badge badge-flexible eslint">` +
+          `${ruleId || 'Fatal'}</${elName}> ${escapeHTML(message)}`
+      } else {
+        ret.text = message
+      }
+      if (linterFix) {
+        ret.fix = linterFix
+      }
+    } catch (err) {
+      ret = await generateInvalidTrace(
```

Comment:
Are you pretty sure that the only kinds of errors which will be caught will be from invalid ranges?  If not, it would be good to check the `err` and re-throw if it is not due to an invalid point.